python/gink/impl/container.py

```diff
@@ -74,6 +74,12 @@ def set_name(self, name: str, *,
         """
         name_property = Muid(-1, -1, Behavior.PROPERTY)
         assert isinstance(name, str), "names must be strings"
+        already_named = self.get_name()
+        if already_named:
+            self._add_entry(
+                key=self._muid, value=deletion, on_muid=name_property,
+                behavior=Behavior.PROPERTY, bundler=bundler, comment=comment
+            )
```

Comment:
I'm not entirely sure if this is the best way to go about this. With the by_name index in the memorystore, if we don't explicitly delete the entry, the old name will remain in the index. Totally open to a better way of doing this.